### PR TITLE
Change Dockerfile to use CentOS instead of Alpine

### PR DESCRIPTION
**Pull request by jenoudet**
Alpine has been unstable and is blocking other upgrades from taking place. This PR returns the Dockerfile to using CentOS.

### CODE REVIEW

File: integration/docker/Dockerfile [hunk 1]
```diff
@@ -77,18 +76,19 @@ ENV TINI_VERSION v0.18.0
...
+    yum install -y java-11-openjdk-devel java-11-openjdk && \
+    yum install -y unzip vim && \
+    yum clean all
 
 # Disable JVM DNS cache (https://github.com/Alluxio/alluxio/pull/9452)
-RUN echo "networkaddress.cache.ttl=0" >> ${JAVA_HOME}/jre/lib/security/java.security
+RUN echo "networkaddress.cache.ttl=0" >> ${JAVA_HOME}/conf/security/java.security
 
 # Add the following for native libraries needed by rocksdb
 ENV LD_LIBRARY_PATH /lib64:${LD_LIBRARY_PATH}
```
Comment (ssz1997) [inline integration/docker/Dockerfile:88]: There are two `ENV LD_LIBRARY_PATH` on line 65 and 88. Let's combine them. The `/usr/local/lib` is for libfuse2 - let's put this information in the comment. People wouldn't know what this is for (for example, me a few months ago)

**Review by ssz1997 — COMMENTED** [review]
Thanks @jenoudet for doing this. Since `Dockerfile-dev` uses CentOS as base image as well, I think we should refactor them into one `Dockerfile`, and potentially get rid of the `dockerfile-common.sh`.

Since this is docker environment, we can use any java version we want. I don't know any known problem with java11, but do know some problems with java8, so I'm ok with just using Java11 and not java8. @LuQQiu What do you think?

**Comment by LuQQiu** [comment]
From my understanding, this PR does two things, change to Centos and change to Java 11? 
Centos change I assume it's seamless to end-users and should be backward compatible.
Java 11 is preferred to Java 8 but it may break the backward compatibility since some users may configure java 8 GC opts which may not work with Java 11.
The other major comment is that, because now Dockerfile and Dockerfile-dev are pretty similar except that Dockerfile-dev has more development dependencies. Can we refactor these two, avoid duplicate codes and ease future developments?

**Comment by Xenorith** [comment]
> The other major comment is that, because now Dockerfile and Dockerfile-dev are pretty similar except that Dockerfile-dev has more development dependencies. Can we refactor these two, avoid duplicate codes and ease future developments?

is it crazy to suggest that the dev image is built on top of the default image by strictly adding new things on top?

**Review by ssz1997 — APPROVED** [review]
LTGM. Thanks!

**Comment by LuQQiu** [comment]
> > The other major comment is that, because now Dockerfile and Dockerfile-dev are pretty similar except that Dockerfile-dev has more development dependencies. Can we refactor these two, avoid duplicate codes and ease future developments?
> 
> is it crazy to suggest that the dev image is built on top of the default image by strictly adding new things on top?

what's the problem with that? it should be only adding new things for development usage

**Comment by jenoudet** [comment]
> > > The other major comment is that, because now Dockerfile and Dockerfile-dev are pretty similar except that Dockerfile-dev has more development dependencies. Can we refactor these two, avoid duplicate codes and ease future developments?
> > 
> > 
> > is it crazy to suggest that the dev image is built on top of the default image by strictly adding new things on top?
> 
> what's the problem with that? it should be only adding new things for development usage

Let's table this discussion for a followup consolidation PR.

**Comment by Xenorith** [comment]
> > > > The other major comment is that, because now Dockerfile and Dockerfile-dev are pretty similar except that Dockerfile-dev has more development dependencies. Can we refactor these two, avoid duplicate codes and ease future developments?
> > > 
> > > 
> > > is it crazy to suggest that the dev image is built on top of the default image by strictly adding new things on top?
> > 
> > 
> > what's the problem with that? it should be only adding new things for development usage
> 
> Let's table this discussion for a followup consolidation PR.

feel free to ignore the crazy guy

**Comment by yuzhu** [comment]
@LuQQiu can we postpone the refactor to a separate PR? We want to unblock the ratis upgrade asap.

**Comment by ssz1997** [comment]
@yuzhu Sure we can merge it.

**Review by yuzhu — APPROVED** [review]

**Comment by yuzhu** [comment]
alluxio-bot, merge this please

**Pull request merged**